# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsChildView.mm
# Commit: df85fe3d7cb6
# Full Hash: df85fe3d7cb6a219c37e0a80d00f51e07783aec1
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 22 - Make nsIDragService::StartDragSession return the nsIDragSession r=rkraesig,win-reviewers,gstoll,geckoview-reviewers,m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D212298
# ==============================================================================

diff -r 384b025f56d3 -r df85fe3d7cb6 widget/cocoa/nsChildView.mm
--- a/widget/cocoa/nsChildView.mm	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/cocoa/nsChildView.mm	Thu Jul 04 07:48:12 2024 +0000
@@ -4126,13 +4126,14 @@
     if (!mDragService) return NSDragOperationNone;
   }
 
+  nsCOMPtr<nsIDragSession> dragSession;
   if (aMessage == eDragEnter) {
     nsIWidget* widget = mGeckoChild;
-    mDragService->StartDragSession(widget);
-  }
-
-  nsCOMPtr<nsIDragSession> dragSession =
-      mDragService->GetCurrentSession(mGeckoChild);
+    dragSession = mDragService->StartDragSession(widget);
+  } else {
+    dragSession = mDragService->GetCurrentSession(mGeckoChild);
+  }
+
   if (dragSession) {
     if (aMessage == eDragOver) {
       // fire the drag event at the source. Just ignore whether it was